app/controllers/survey_controller.rb

```diff
@@ -17,7 +17,7 @@ def create_question
         q.save!
       end
     end
-    @question.save!
+    @question.save(:validate => false)
```

Comment:
It prevents the validation checks from happening until all parts of the question have been assembled (and then it runs the validation checks).